src/contexts/modal/ModalContext.tsx

```diff
@@ -109,6 +113,15 @@ function ModalProvider({ children }: Props) {
     [showModal, hideModal]
   );
 
+  // close modal on route change
+  const route = useStabilizedRouteTransitionKey();
+  useEffect(() => {
+    if (isModalOpenRef.current) {
+      // bypass onClose as to not navigate the user back mid-route change
+      hideModal(true);
+    }
+  }, [route, hideModal]);
```

Comment:
1. hide modal when navigating to a new page. should be default behavior, since keeping a modal open while re-routing is weird. this fixes a related error where going to the Community Page from the detail modal crashed, because the modal remained open, and it was looking at URL params to continue to display NFT detail stuff.
2. also added a `bypassOnClose` prop to simply hide the modal without triggering the `onClose`. this was because the NFT Detail Modal was programmed to go "back" when it was closed, which collided with the redirect to the community page